src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java

```diff
@@ -124,7 +124,7 @@ private static void destroy(String container, Launcher launcher, Node node, EnvV
 
         @Override public boolean start() throws Exception {
             EnvVars envReduced = new EnvVars(env);
-            EnvVars envHost = computer.getEnvironment();
+            EnvVars envHost = computer.buildEnvironment(TaskListener.NULL);
```

Comment:
Actually no this is definitely wrong. `envHost` is deliberately set to native environment variables of the computer, which is what `getEnvironment` returns. `buildEnvironment` returns totally unrelated—Jenkins-specific—variables. So whatever your issue is, this is not the right fix.